h2o-algos/src/main/java/hex/tree/TreeHandler.java

```diff
@@ -62,10 +63,106 @@ else if (!(possibleModel instanceof SharedTreeModel) && !(possibleModel instance
         args.nas = treeProperties._nas;
         args.levels = treeProperties.levels;
         args.predictions = treeProperties._predictions;
+        args.language_tree_representation = treeProperties._languageTreeRepresentation;
+        args.language_path_representations = treeProperties._languagePathsRepresentations;
 
         return args;
     }
 
+
+    private static String getLanguageRepresentation(SharedTreeSubgraph sharedTreeSubgraph) {
+        return getNodeRepresentation(sharedTreeSubgraph.rootNode, "", 0);
+    }
+
+    private static String getNodeRepresentation(SharedTreeNode node, String languageRepresentation, int padding) {
+        if (node.getLeftChild() != null || node.getRightChild() != null) {
+            if (padding != 0)
+                languageRepresentation += getNewPaddedLine(padding);
+            languageRepresentation += getConditionLine(node);
+            if (node.getRightChild() != null) {
+                languageRepresentation += getNewPaddedLine(padding);
+                languageRepresentation = getNodeRepresentation(node.getRightChild(), languageRepresentation, padding + 1);
+            }
+            languageRepresentation += getNewPaddedLine(padding);
+            languageRepresentation += getElseLine(node);
```

Comment:
I thnk getLeftChild is never null if it were, the else branch would be empty and the "code" invalid

Reply:
fixed